aptos-move/move-examples/marketplace/sources/fee_schedule.move

```diff
@@ -240,6 +301,91 @@ module marketplace::fee_schedule {
         emit_mutation_event(signer::address_of(&fee_schedule_signer), updated_resource);
     }
 
+    public(friend) fun emit_bid_event(
+        marketplace: Object<FeeSchedule>,
+        listing: address,
+        new_bidder: address,
+        new_bid: u64,
+        new_end_time: u64,
+        previous_bidder: Option<address>,
+        previous_bid: Option<u64>,
+        previous_end_time: u64,
+    ) acquires MarketplaceEventsV1 {
+        let marketplace_events = get_marketplace_events_v1(marketplace);
+        event::emit_event<BidEvent>(&mut marketplace_events.bid_events, BidEvent {
+            listing,
+            new_bidder,
+            new_bid,
+            new_end_time,
+            previous_bidder,
+            previous_bid,
+            previous_end_time,
+        });
+    }
+
+    public(friend) fun emit_purchase_event(
+        marketplace: Object<FeeSchedule>,
+        listing: address,
+        purchaser: address,
+        price: u64,
+        commission: u64,
+        royalties: u64
+    ) acquires MarketplaceEventsV1 {
+        let marketplace_events = get_marketplace_events_v1(marketplace);
+        event::emit_event<PurchaseEvent>(&mut marketplace_events.purchase_events, PurchaseEvent {
+            listing,
+            purchaser,
+            price,
+            commission,
+            royalties,
+        });
+    }
+
+    public(friend) fun emit_collection_offer_event(
+        marketplace: Object<FeeSchedule>,
+        offer: address,
+        seller: address,
+        price: u64,
```

Comment:
I'm gonna assume this is price per token? Can we include # of tokens in the event?

Reply:
Yeah the marketplace is all non-fungible, but for collection bids, there can be multiple tokens filled. In this case, it looks like we emit 1 event per token filled in a collection bid, so token amount here will always be 1.